### PR TITLE
Add [Debian] package version support

### PR DESCRIPTION
**Pull request by scop**
https://wiki.debian.org/Services/api.ftp-master.debian.org

### CODE REVIEW

File: services/debian/debian.service.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      throw new NotFound()
+    }
+    // Distribution can change compared to request, for example requesting
+    // "stretch" may map to "stable" in results, so can't use value from
+    // request as is. Just take the first one instead.
+    for (const resultdist of Object.keys(data[0][packageName])) {
+      const versions = Object.keys(data[0][packageName][resultdist])
+      return renderVersionBadge({ version: latest(versions) })
+    }
+    throw new NotFound()
```
Comment (chris48s) [inline services/debian/debian.service.js]: Is this line reachable?

File: services/debian/debian.service.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      schema,
+      url: `https://api.ftp-master.debian.org/madison?f=json&s=${distribution}&package=${packageName}`,
+    })
+    if (!data.length) {
+      throw new NotFound()
+    }
+    // Distribution can change compared to request, for example requesting
+    // "stretch" may map to "stable" in results, so can't use value from
+    // request as is. Just take the first one instead.
+    for (const resultdist of Object.keys(data[0][packageName])) {
```
Comment (chris48s) [inline services/debian/debian.service.js]: I'm not familiar with this API. Is there any valid case where `data.length` can be >1?

File: services/debian/debian.tester.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+'use strict'
+
+const { isVPlusDottedVersionNClauses } = require('../test-validators')
+const t = (module.exports = require('../tester').createServiceTester())
+
+t.create('Debian package (default distribution, valid)')
+  .get('/apt.json')
+  .expectBadge({ label: 'debian', message: isVPlusDottedVersionNClauses })
```
Comment (chris48s) [inline services/debian/debian.tester.js]: Its reasonably common for Debian packages to have some kind of suffix on the version e.g: https://api.ftp-master.debian.org/madison?f=json&s=testing&package=python-dev Is `apt` always going to be a version without a suffix? If not, can we go with `isVPlusDottedVersionNClausesWithOptionalSuffix` to make the test less brittle?

File: services/debian/debian.service.js (hunk 1 above)
Comment (scop) [reply services/debian/debian.service.js]: It is, if a package is returned without any versions. I suppose that shouldn't happen, but if it does, "not found" seems appropriate behavior to me.

File: services/debian/debian.service.js (hunk 2 above)
Comment (scop) [reply services/debian/debian.service.js]: Probably not. Would you like a case where it is handled somehow differently?

File: services/debian/debian.tester.js (hunk 3 above)
Comment (scop) [reply services/debian/debian.tester.js]: I don't think there are any such guarantees, so done. See also #3378

**Comment by scop** [comment]
A couple of improvements I thought of but didn't get around to implementing as I didn't immediately spot what would be the best way to do them:
- Maybe mark versions containing `~` as prereleases -- "maybe" because that's what they usually are, but not strictly always; can be backports and such. 
- Implementing epoch aware ordering, possibly also stripping epochs from displayed versions; ref https://www.debian.org/doc/debian-policy/ch-controlfields.html#version, example https://packages.debian.org/stretch/bash-completion

File: services/debian/debian.service.js (hunk 2 above)
Comment (chris48s) [reply services/debian/debian.service.js]: If we're getting back arrays which should only contain zero or one elements (and we don't know what do do if we get more), can we validate that in the schema.

I can't remember off the top of my head, but I think chaining `.single()` without `required()` will pass zero or one elements. If not, chaining `.min(0).max(1)` will do the job.

Comment (scop) [reply services/debian/debian.service.js]: I think `.single()` is the other way around (get single item as if it was an array). `.max(1).required()` seems to do the right thing.

**Comment by paulmelnikow** [comment]
This is a really nice addition! Thank you!

**Pull request merged**